docs/releases/2.0.5.txt

```diff
@@ -23,3 +23,8 @@ Bugfixes
   ``values_list()`` after combining an annotated and unannotated queryset with
   ``union()``, ``difference()``, or ``intersection()`` crashed due to mismatching
   columns (:ticket:`29286`).
+
+* Fixed issue where the ``allow_relation()`` check was never enforced.
```

Comment:
I don't believe this qualifies for a backport per our [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions). Especially given the possibility of breaking existing code. Let's target this for Django 2.1 -- a note can be added in the "backwards incompatible changes" section of the release notes.